sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb

```diff
@@ -0,0 +1,239 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training Data\n",
+    "\n",
+    "The MNIST dataset has been loaded to public S3 buckets ``sagemaker-sample-data-<REGION>`` under prefix ``tensorflow/mnist``. There are four ``.npy`` file under this prefix: ``train_data.npy``, ``eval_data.npy``, ``train_labels.npy`` and ``eval_labels.npy``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_data_uri = 's3://sagemaker-sample-data-us-west-2/tensorflow/mnist'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Construct a script for distributed training \n",
+    "\n",
+    "The training script was adapted from TensorFlow's official [CNN MNIST example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py). We have modified to handle a ``model_dir`` parameter passed in by SageMaker. This is a S3 path which can be used for data sharing during distributed training and checkpoint and or model persistence. In addtion we have also added a argument parsing function to handle processing training related variables.\n",
+    "\n",
+    "At the end of the training job we also added a step to export the trained model to the path stored in the environment variable ``SM_MODEL_DIR`` which always points to ``/opt/ml/model``. This is critical because SageMaker uploads all the model artifact in this folder to S3 at end of training.\n",
+    "\n",
+    "Here is the entire script:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!cat 'mnist.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a training job using the sagemaker.TensorFlow estimator\n",
+    "\n",
+    "The ``sagemaker.TensorFlow`` estimator handles locating the script mode container uploading your script to a S3 location and creating a SageMaker training job. Let's call out couple important paramters here:\n",
+    "\n",
+    "* ``py_version`` is set to ``'py3'`` which implys we are using script mode since legacy mode only supports Python2.7. Although we don't recommand using Python2 but you can set ``py_version`` to ``'py2'`` and ``script_mode`` to ``True`` to use script mode with Python2.\n",
+    "\n",
+    "* ``distributions`` is used to configure distributed training setup. It's only required if you are doing distributed training either across a cluster of instances or across multiple GPUs. Here we are using parameter server as the distributed training schema. SageMaker training jobs are running on homogeneous clusters and to make parameter server more performant we run a parameter server on every instance in the training job cluster. So there is no need to specify the number of paramter servers to launch.\n",
```

Comment:
* s/configure distributed/configure the distributed
* s/only required if/required only if
* s/parameter server/parameter servers
* s/are running/run
* s/paramter/parameter

for the last two sentences, I'd put a period after "homogeneous clusters" and make the rest of it one sentence, i.e. "To make the parameter server...cluster, so there is no..."